src/hotspot/share/opto/divnode.cpp

```diff
@@ -1379,37 +1397,49 @@ const Type* ModLNode::Value(PhaseGVN* phase) const {
 
 
 //=============================================================================
-//------------------------------Value------------------------------------------
-const Type* ModFNode::Value(PhaseGVN* phase) const {
-  // Either input is TOP ==> the result is TOP
-  const Type *t1 = phase->type( in(1) );
-  const Type *t2 = phase->type( in(2) );
-  if( t1 == Type::TOP ) return Type::TOP;
-  if( t2 == Type::TOP ) return Type::TOP;
+//------------------------------Idealize---------------------------------------
+Node *UModLNode::Ideal(PhaseGVN *phase, bool can_reshape) {
+  return unsigned_mod_ideal<TypeLong, julong>(phase, can_reshape, this);
+}
 
-  // Either input is BOTTOM ==> the result is the local BOTTOM
-  const Type *bot = bottom_type();
-  if( (t1 == bot) || (t2 == bot) ||
-      (t1 == Type::BOTTOM) || (t2 == Type::BOTTOM) )
-    return bot;
+const Type* UModLNode::Value(PhaseGVN* phase) const {
+  return unsigned_mod_value<TypeLong, julong, jlong>(phase, this);
+}
+
+Node* ModFNode::Ideal(PhaseGVN* phase, bool can_reshape) {
```

Comment:
Can you quickly say why you congerted this from a `Value` to an `Ideal` method? I guess it is because before it used to be a simple `Node` with a single output, but now it is a `Call` with multiple outputs... Ok makes sense.